Make hyperv write operations compile-time optional
--- linux.orig/tools/hv/hv_kvp_daemon.c
+++ linux/tools/hv/hv_kvp_daemon.c
@@ -1922,6 +1922,7 @@ reopen_kvp_fd:
 
 			break;
 
+#ifdef KVP_ENABLE_WRITE
 		case KVP_OP_SET_IP_INFO:
 			kvp_ip_val = &hv_msg->body.kvp_ip_val;
 			if_name = kvp_get_if_name(
@@ -1949,6 +1950,7 @@ reopen_kvp_fd:
 					hv_msg->body.kvp_set.data.value_size))
 					hv_msg->error = HV_S_CONT;
 			break;
+#endif
 
 		case KVP_OP_GET:
 			if (kvp_get_value(pool,
@@ -1959,12 +1961,14 @@ reopen_kvp_fd:
 					hv_msg->error = HV_S_CONT;
 			break;
 
+#ifdef KVP_ENABLE_WRITE
 		case KVP_OP_DELETE:
 			if (kvp_key_delete(pool,
 					hv_msg->body.kvp_delete.key,
 					hv_msg->body.kvp_delete.key_size))
 					hv_msg->error = HV_S_CONT;
 			break;
+#endif
 
 		default:
 			break;
