--- linux.orig/tools/hv/hv_get_dhcp_info.sh
+++ linux/tools/hv/hv_get_dhcp_info.sh
@@ -1,19 +1,11 @@
 #!/bin/bash
 # SPDX-License-Identifier: GPL-2.0
 
-# This example script retrieves the DHCP state of a given interface.
-# In the interest of keeping the KVP daemon code free of distro specific
-# information; the kvp daemon code invokes this external script to gather
-# DHCP setting for the specific interface.
-#
 # Input: Name of the interface
 #
 # Output: The script prints the string "Enabled" to stdout to indicate
 #	that DHCP is enabled on the interface. If DHCP is not enabled,
 #	the script prints the string "Disabled" to stdout.
-#
-# Distributions may need to adapt or replace this script for their
-# preferred network configuration system.
 
 # Report status based on result of previous command
 report() {
@@ -85,8 +77,18 @@ check_network_scripts() {
     fi
 }
 
-check_network_manager "$1" ||
-check_systemd_networkd "$1" ||
-check_ifupdown "$1" ||
-check_network_scripts "$1" ||
+check_pexip_flag() {
+    if [ -f "/run/pexip-net-info" ]; then
+	true
+	report
+    else
+	return 1
+    fi
+}
+
+#check_network_manager "$1" ||
+#check_systemd_networkd "$1" ||
+#check_ifupdown "$1" ||
+#check_network_scripts "$1" ||
+check_pexip_flag "$1" ||
 report
--- linux.orig/tools/hv/hv_get_dns_info.sh
+++ linux/tools/hv/hv_get_dns_info.sh
@@ -1,13 +1,3 @@
 #!/bin/sh
 
-# This example script parses /etc/resolv.conf to retrive DNS information.
-# In the interest of keeping the KVP daemon code free of distro specific
-# information; the kvp daemon code invokes this external script to gather
-# DNS information.
-# This script is expected to print the nameserver values to stdout.
-# Each Distro is expected to implement this script in a distro specific
-# fashion. For instance on Distros that ship with Network Manager enabled,
-# this script can be based on the Network Manager APIs for retrieving DNS
-# entries.
-
 exec awk '/^nameserver/ { print $2 }' /etc/resolv.conf 2>/dev/null
