Index: os-linux/include/uapi/linux/types.h
===================================================================
--- os-linux.orig/include/uapi/linux/types.h
+++ os-linux/include/uapi/linux/types.h
@@ -13,10 +13,12 @@
 
 #include <linux/posix_types.h>
 
+#ifndef __INTEL_COMPILER
 #ifdef __SIZEOF_INT128__
 typedef __signed__ __int128 __s128 __attribute__((aligned(16)));
 typedef unsigned __int128 __u128 __attribute__((aligned(16)));
 #endif
+#endif
 
 /*
  * Below are truly Linux-specific types that should never collide with
